File: akka-samples/scala-samples/src/main/scala/actors/context/ContextPrinter.scala
package actors.context

import akka.actor.typed.scaladsl.Behaviors
import akka.actor.typed.{ActorRef, ActorSystem, Behavior}
import actors.context.Printer.PrintMessage

import java.util.Calendar

object ContextPrinter {
  def main(args: Array[String]): Unit = {
    val system = ActorSystem[PrintMessage](Printer(), "simple-printer")
    system ! PrintMessage("Hello there!")
  }
}

private object Printer {
  final case class PrintMessage(text: String)

  def date(): Behavior[PrintMessage] = Behaviors.setup { context =>
    println(Calendar.getInstance().getTime)
    Behaviors.stopped
  }

  def apply(): Behavior[PrintMessage] = {
    Behaviors.setup { context =>
      val reference: ActorRef[PrintMessage] = context.spawn(date(), "datetime-actor")

      Behaviors.receive[PrintMessage] { (context, message) =>
        println(message.text)
        context.log.info(s"Received message: ${message.text}")

        Behaviors.same
      }
    }
  }
}

File: akka-samples/java-samples/src/main/java/actors/context/ContextPrinter.java
package actors.context;

import akka.actor.typed.ActorRef;
import akka.actor.typed.ActorSystem;
import akka.actor.typed.Behavior;
import akka.actor.typed.javadsl.AbstractBehavior;
import akka.actor.typed.javadsl.ActorContext;
import akka.actor.typed.javadsl.Behaviors;
import akka.actor.typed.javadsl.Receive;

import java.util.Calendar;

class ContextPrinter extends AbstractBehavior<ContextPrinter.PrintMessage> {
    static final class PrintMessage {
        public final String text;

        public PrintMessage(String text) {
            this.text = text;
        }
    }

    public ContextPrinter(ActorContext<PrintMessage> context) {
        super(context);
        ActorRef<PrintMessage> reference = context.spawn(date(), "datetime-actor");
    }

    private Behavior<PrintMessage> date() {
        return Behaviors.setup(context -> {
            System.out.println(Calendar.getInstance().getTime());
            return Behaviors.stopped();
        });
    }

    public static Behavior<PrintMessage> create() {
        return Behaviors.setup(ContextPrinter::new);
    }

    @Override
    public Receive<PrintMessage> createReceive() {
        return newReceiveBuilder().onMessage(PrintMessage.class, (message) -> {
            System.out.println(message.text);
            getContext().getLog().info(String.format("Received message: %s", message.text));

            return Behaviors.same();
        }).build();
    }

    public static void main(String[] args) {
        final ActorSystem<PrintMessage> system = ActorSystem.create(ContextPrinter.create(), "simple-printer");
        system.tell(new PrintMessage("Hello there!"));
    }
}